### PR TITLE
[0.3.x] CAL-395 Set gib.failOnError=false

### PR DESCRIPTION
**Pull request by LinkMJB**
See: https://github.com/vackosar/gitflow-incremental-builder/issues/15#issuecomment-318679437

#### What does this PR do?
Updates default GIB behavior to perform a full build if it loses git context
#### Who is reviewing it? 
@oconnormi @lessarderic @shaundmorris @vinamartin 
#### Select relevant component teams: 
@codice/continuous-integration 
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@lessarderic
@shaundmorris
#### How should this be tested? (List steps with links to updated documentation)
Perform PR build on CI environment
Perform manual GIB enabled build (use options: `-Dgib.enabled=true -Dgib.referenceBranch=refs/remotes/origin/0.3.x`)
#### Any background context you want to provide?
See: https://trello.com/c/7iBsxkdi
#### What are the relevant tickets?
- https://codice.atlassian.net/projects/CAL/issues/CAL-395
#### Checklist:
- [N/A] Documentation Updated
- [N/A] Update / Add Unit Tests
- [N/A] Update / Add Integration Tests

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by LinkMJB** [comment]
build now

**Comment by LinkMJB** [comment]
Last build failed due to unrelated OWASP finding, supposed to be patched upstream by DDF.

build now

**Comment by LinkMJB** [comment]
Last build failed due to unrelated OWASP finding, supposed to be patched upstream by DDF.

build now

**Comment by LinkMJB** [comment]
Last build failed due to unrelated OWASP finding, supposed to be patched upstream by DDF.

build now

File: pom.xml [hunk 1]
```diff
@@ -202,7 +203,7 @@
             <extension>
                 <groupId>com.vackosar.gitflowincrementalbuilder</groupId>
                 <artifactId>gitflow-incremental-builder</artifactId>
-                <version>3.1</version>
+                <version>3.4</version>
```
Comment (brendan-hofmann) [inline pom.xml]: This is just bumped to match master.

Comment (LinkMJB) [reply pom.xml]: @brendan-hofmann also the failOnError argument isn't valid/supported in 3.1 -- it caused the build to fail without upgrading.

**Pull request merged**